tensorflow/core/kernels/data/range_dataset_op_test.cc

```diff
@@ -37,73 +38,170 @@ class RangeDatasetOpTest : public DatasetOpsTestBase {
   }
 };
 
-struct TestCase {
-  int64 start;
-  int64 stop;
-  int64 step;
-  std::vector<Tensor> expected_outputs;
-  DataTypeVector expected_output_dtypes;
-  std::vector<PartialTensorShape> expected_output_shapes;
-  int64 expected_cardinality;
-  std::vector<int> breakpoints;
+class RangeDatasetParams : DatasetParams {
+ public:
+  RangeDatasetParams(int64 start, int64 stop, int64 step,
+                     DataTypeVector output_dtypes,
+                     std::vector<PartialTensorShape> output_shapes)
+      : DatasetParams(std::move(output_dtypes), std::move(output_shapes)),
+        start(
+            DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {start})),
+        stop(DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {stop})),
+        step(DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {step})) {
+  }
+
+  Status MakeInputs(gtl::InlinedVector<TensorValue, 4>* inputs) override {
+    *inputs = {TensorValue(&start), TensorValue(&stop), TensorValue(&step)};
+    return Status::OK();
+  }
+
+  Tensor start;
+  Tensor stop;
+  Tensor step;
 };
 
-TestCase PositiveStepTestCase() {
-  return {/*start*/ 0,
-          /*stop*/ 10,
-          /*step*/ 3,
-          /*expected_outputs*/
+RangeDatasetParams PositiveStepRangeDataset() {
+  return {/*start=*/0,
+          /*stop=*/10,
+          /*step=*/3,
+          /*output_dtypes=*/{DT_INT64},
+          /*output_shapes=*/{PartialTensorShape({})}};
+}
+
+RangeDatasetParams NegativeStepRangeDataset() {
+  return {/*start=*/10,
+          /*stop=*/0,
+          /*step=*/-3,
+          /*output_dtypes=*/{DT_INT64},
+          /*output_shapes=*/{PartialTensorShape({})}};
+}
+
+RangeDatasetParams ZeroStepRangeDataset() {
+  return {/*start=*/10,
+          /*stop=*/0,
+          /*step=*/0,
+          /*output_dtypes=*/{DT_INT64},
+          /*output_shapes=*/{PartialTensorShape({})}};
+}
+
+GetNextTestCase<RangeDatasetParams> GetNextTestCase1() {
```

Comment:
move the definition of test cases just before the unit test for which they are used